internal/authtoken/repository.go

```diff
@@ -290,29 +292,147 @@ func (r *Repository) ValidateToken(ctx context.Context, id, token string, opt ..
 	return retAT, nil
 }
 
-// ListAuthTokens lists auth tokens in the given scopes and supports the
-// WithLimit option.
-func (r *Repository) ListAuthTokens(ctx context.Context, withScopeIds []string, opt ...Option) ([]*AuthToken, error) {
-	const op = "authtoken.(Repository).ListAuthTokens"
+// listAuthTokens lists auth tokens in the given scopes and supports the
+// WithLimit and WithStartPageAfterItem options.
+func (r *Repository) listAuthTokens(ctx context.Context, withScopeIds []string, opt ...Option) ([]*AuthToken, time.Time, error) {
+	const op = "authtoken.(Repository).listAuthTokens"
 	if len(withScopeIds) == 0 {
-		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing scope id")
+		return nil, time.Time{}, errors.New(ctx, errors.InvalidParameter, op, "missing scope id")
 	}
 	opts := getOpts(opt...)
 
-	// use the view, to bring in the required account columns. Just don't forget
-	// to convert them before returning them
-	var atvs []*authTokenView
-	if err := r.reader.SearchWhere(ctx, &atvs, "auth_account_id in (select public_id from auth_account where scope_id in (?))", []any{withScopeIds}, db.WithLimit(opts.withLimit)); err != nil {
-		return nil, errors.Wrap(ctx, err, op)
+	limit := r.limit
+	if opts.withLimit != 0 {
+		// non-zero signals an override of the default limit for the repo.
+		limit = opts.withLimit
+	}
+
+	whereScopeIds := "('" + strings.Join(withScopeIds, "', '") + "')"
```

Comment:
@johanbrandhorst I was between this method and looping through `withScopeIds` to turn them into `@scope_id_%d` args, do you have a preference?

Reply:
Definitely using args, this here is much more vulnerable to SQL injection. I think the `scope_id in` part should also be in here somewhere.

Reply:
Credential libraries has an example of this: https://github.com/hashicorp/boundary/pull/4090/files#diff-edc9e7d38750a08b4a30397ed90cfe1ebb2398c64e9ebb2d7009356d9e2f4996